lib/albacore/task_types/build.rb

```diff
@@ -189,25 +190,10 @@ def heuristic_executable
 
     	  msb = "msbuild_not_found"
     	  maxVersion = -1
-    	  begin
-    		Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\MSBuild\ToolsVersions') do |toolsVersion|
-    		  toolsVersion.each_key do |key|
-    			begin
-    			  versionKey = toolsVersion.open(key)
-    			  version = key.to_i
-    			  if maxVersion < version
-    				maxVersion = version
-    				msb = "#{versionKey['MSBuildToolsPath']}\\msbuild.exe"
-    			  end
-    			rescue
-    			  error "failed to open #{key}"
-    			end
-    		  end
-    		end
-    	  rescue
-    		error "failed to open HKLM\\SOFTWARE\\Microsoft\\MSBuild\\ToolsVersions"
-    	  end
-
+        versions = Albacore.find_msbuild_versions
+        if versions.any?
+          msb = versions.max[1]
```

Comment:
[http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-max](Enumerable max) probably:
[[14, "PATH14"], [12, "PATH12"], [11, "PATH11"]].max 
has the same behaviour as 
{14=>"PATH14", 12=>"PATH12", 11=>"PATH11"}.max